Doc/extending/extending.rst

```diff
@@ -341,6 +345,17 @@ accept a third ``PyObject *`` parameter which will be a dictionary of keywords.
 Use :c:func:`PyArg_ParseTupleAndKeywords` to parse the arguments to such a
 function.
 
+The fourth entry is the function's docstring. Note the
+``"system(command)\n--\n\n"`` at the beginning. Since C extension functions
+take a fixed number of arguments---e.g. two for ``METH_VARARGS``---the Python
+interpreter has no way of knowing the function's signature until the function
+is called. However, by making the docstring start with something of the form
+``"myfunc(arg1, arg2='my_default', arg3=17)\n--\n\n"``, the signature will be
+constructed and the remainder of the docstring will be used as the function's
+``__doc__`` attribute. This can be useful for generating nice messages with the
+:func:`help` function or for getting the signature in object form with the
+:func:`inspect.signature` function.
```

Comment:
```suggestion
The fourth entry is the function's docstring. Note the
``"system(command)\n--\n\n"`` at the beginning. Since C extension functions
take a fixed number of arguments (for instance, two for ``METH_VARARGS``) the Python
interpreter has no way of knowing the function's signature until the function
is called. However, by making the docstring start with something of the form
``"myfunc(arg1, arg2='my_default', arg3=17)\n--\n\n"``, the signature will be
constructed and the remainder of the docstring will be used as the function's
``__doc__`` attribute. This is typically useful for generating nice messages with
:func:`help` or for getting the signature object with :func:`inspect.signature`.
```